docs/SalesforceMultiObjects-batchsource.md

```diff
@@ -0,0 +1,121 @@
+# Salesforce Batch Multi Source
+
+
+Description
+-----------
+This source reads multiple sObjects from Salesforce. The data which should be read is specified using list of sObjects 
+and incremental or range date filters. The source will output a record for each row in the SObjects it reads, 
+with each record containing an additional field that holds the name of the SObject the record came from. 
+In addition, for each SObject that will be read, this plugin will set pipeline arguments where the key is 'multisink.[SObjectName]' 
+and the value is the schema of the SObject.
+
+Configuration
+-------------
+
+**Reference Name:** Name used to uniquely identify this source for lineage, annotating metadata, etc.
+
+**Username:** Salesforce username.
+
+**Password:** Salesforce password.
+
+**Client Id:** Application Client Id. This is also called "Consumer Key" in Salesforce UI.
+To create Client Id user needs to create a connected application in Salesforce first.
+
+**Client Secret:** Application Client Secret. This is also called "Consumer Secret" in Salesforce UI.
+To create Client Secret user needs to create a connected application in Salesforce first.
+
+**Login Url:** Salesforce OAuth2 login url.
+
+**White List**: List of SObjects to replicate from Salesforce. By default all SObjects will be white listed. 
+For each white listed SObject, SOQL query will be generated: `select <FIELD_1, FIELD_2, ..., FIELD_N> from ${sObjectName}`.
+
+**Black List**: List of SObjects NOT to replicate from Salesforce. By default NONE of SObjects will be black listed.
+
+**Datetime Filter:** SObject query datetime filter is applied to system field `LastModifiedDate` to allow incremental 
```

Comment:
OK let's open a jira and fix separately.

After reading the doc again I understand better, this is only used when an SObject is given and not for SOQL. I think the confusing part is that users can either specify a DatetimeFilter (which is really just specifying the start time), or they can specify a duration and offset.

I think the confusion can be removed if instead of these three, we have a 'start' and 'end' property. If 'start' is not specified, it reads from the start of time. If 'end' is not specified, it reads up until the logical start time of the pipeline (context.getLogicalStartTime()). This way, 'start' replaces the current DatetimeFilter, and it is probably easier to understand than duration and offset.

Reply:
Created Jira for the fix: https://issues.cask.co/browse/CDAP-15320